egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -0,0 +1,310 @@
+        amount of examples in each language.
+        It works based on sampling from a
+        discrete distribution, where it returns i
+        with prob(i) = (num_egs in lang(i)/ tot_egs).
+        tot_egs is sum of lang_len.
+    """
+    assert(tot_egs > 0)
+    rand_int = random.randint(0, tot_egs - 1)
+    count = 0
+    for l in range(len(lang_len)):
+        if random_selection:
+            if rand_int > count and rand_int <= (count + lang_len[l]):
+                rand_lang = l
+                break
+            else:
+                count += lang_len[l]
+        else:
+            if (lang_len[l] > 0):
+                rand_lang = l
+                break
+    assert(rand_lang >= 0 and rand_lang < len(lang_len))
+    return rand_lang
+
+def read_lang2weight(lang2w_file):
+    """ Read lang2weight file where each line is a 2-tuple of
+        (lang-num lang-weight) and return array of lang-weight
+        as lang2w where lang2w[i] is the weight for language i.
+    """
+    f = open(lang2w_file, "r")
+    if f is None:
+        raise Exception("Error opening file {0}".format(lang2w_file))
+    lang2w = []
+    for line in f:
+        a = line.split()
+        if len(a) != 2:
+            raise Exception("bad line {0} in file.".format(line))
+        lang2w.append(int(a[1]))
+    f.close()
+    return lang2w
+
+def  process_multilingual_egs(args):
+    args = get_args()
+    random.seed(args.seed)
+    num_langs = args.num_langs
+    rand_select = args.random_lang
+
+    # read egs.scp for input languages
+    scp_lists = args.egs_scp_lists.split()
+    assert(len(scp_lists) == num_langs)
+
+    scp_files = [open(scp_lists[lang], 'r') for lang in range(num_langs)]
+
+    lang2len = [0] * num_langs
+    for lang in range(num_langs):
+        lang2len[lang] = sum(1 for line in open(scp_lists[lang]))
+        logger.info("Number of examples for language {0} "
+                    "is {1}.".format(lang, lang2len[lang]))
+
+    # If weights are not provided, the weights are 1.0.
+    if args.lang2weight is None:
+        lang2weight = [ 1.0 ] * num_langs
+    else:
+        lang2weight = read_lang2weight(args.lang2weight)
+        assert(len(lang2weight) == num_langs)
+
+    if not os.path.exists("{0}/temp".format(args.egs_dir)):
+        os.makedirs("{0}/temp".format(args.egs_dir))
+    num_lang_file = open("{0}/info/{1}num_tasks".format(args.egs_dir, args.prefix), "w")
+    print("{0}".format(num_langs), file=num_lang_file)
+
+    # Each element of all_egs (one per num_archive * num_jobs) is
+    # an array of 3-tuples (lang-id, local-start-egs-line, num-egs)
+    all_egs = []
+    lang_len = lang2len[:]
+    # total num of egs in all languages
+    tot_num_egs = sum(lang2len[i] for i in range(len(lang2len)))
+    num_archives = max(1, min(args.max_archives, tot_num_egs / args.samples_per_iter))
+
+    num_arch_file = open("{0}/info/{1}num_archives".format(
+        args.egs_dir,
+        args.prefix), "w")
```

Comment:
"w" is an argument to open. So it should be in a separate line and same column as the left parenthesis of open.